backend/src/services/service-token/service-token-service.ts

```diff
@@ -185,11 +189,39 @@ export const serviceTokenServiceFactory = ({
     return { ...serviceToken, lastUsed: new Date(), orgId: project.orgId };
   };
 
+  const notifyExpiredTokens = async () => {
+    const appCfg = getConfig();
+
+    const expiredTokens = await serviceTokenDAL.findExpiredTokens();
```

Comment:
batching just the find operation won't really change anything because we're still processing everything all at once here

ideally, the flow would be:
1. fetch first 500 "almost" expiring tokens
2. send email for the first 500
3. fetch second 500
4. send email for second 500
...